snowflake_data_profiler/profiling/profiler.py

```diff
@@ -76,7 +86,8 @@ def get_profile_results(data):
 
 
 def do_profile():
-    pd_df = connect_to_snowflake()
+    conn = establish_connection()
+    pd_df = create_cursor(conn)
     return get_profile_results(pd_df)
```

Comment:
If our functions have been designed well, we'll achieve the following here:
1. our code will be extremely readable. A brand new reader should understand what's happening here
2. our code will be easily testable with limited side effects in our code that we need to handle with mocks
3. our code will be easily debuggable. If we have an issue, it'll be easy to identify where the issue is coming from.

I often start with pseudo code in situations like this:

```python
def get_snowflake_connection(args):
    ...
    return sf_connection 

def get_snowflake_data(sf_connection, args):
    ...
    return sf_data_df

def get_profile_report(sf_data_df, args):
    ...
    return profile_report

def main(args):
  sf_connection = get_snowflake_connection(args)
  sf_data_df = get_snowflake_data(sf_connection, args)
  profile_report = get_profile_report(sf_data_df, args)
```